pandas/tests/io/excel/test_writers.py

```diff
@@ -233,6 +233,23 @@ def test_read_excel_parse_dates(self, ext):
             )
             tm.assert_frame_equal(df, res)
 
+    def test_excel_table_roundtrip_indexname(self, ext):
+        if ext == ".xls":
+            pytest.skip()
+        df = DataFrame(np.random.randn(10, 4))
+
+        df.columns = df.columns.map(str)
+        df.index.name = "foo"
+
+        with ensure_clean(ext) as pth:
+            df.to_excel(pth, header=True, table="Table1")
+
+            xf = ExcelFile(pth)
```

Comment:
Should use this as a context manager

Reply:
Are you sure? This style is consistent with all the other tests